Core/include/Acts/Vertexing/AdaptiveMultiVertexFitter.ipp

```diff
@@ -256,36 +265,36 @@ Acts::Result<void> Acts::
         State& state, const linearizer_t& linearizer,
         const VertexingOptions<input_track_t>& vertexingOptions) const {
   for (auto vtx : state.vertexCollection) {
-    VertexInfo<input_track_t>& currentVtxInfo = state.vtxInfoMap[vtx];
+    VertexInfo<input_track_t>& vtxInfo = state.vtxInfoMap[vtx];
 
     const std::shared_ptr<PerigeeSurface> vtxPerigeeSurface =
         Surface::makeShared<PerigeeSurface>(
-            VectorHelpers::position(state.vtxInfoMap[vtx].oldPosition));
+            VectorHelpers::position(vtxInfo.oldPosition));
 
-    for (const auto& trk : currentVtxInfo.trackLinks) {
+    if (vtxInfo.relinearize) {
+      vtxInfo.linPoint = vtxInfo.oldPosition;
+    }
+
+    for (const auto& trk : vtxInfo.trackLinks) {
       auto& trkAtVtx = state.tracksAtVerticesMap.at(std::make_pair(trk, vtx));
 
       // Set trackWeight for current track
-      double currentTrkWeight = m_cfg.annealingTool.getWeight(
+      trkAtVtx.trackWeight = m_cfg.annealingTool.getWeight(
           state.annealingState, trkAtVtx.vertexCompatibility,
           collectTrackToVertexCompatibilities(state, trk));
-      trkAtVtx.trackWeight = currentTrkWeight;
 
       if (trkAtVtx.trackWeight > m_cfg.minWeight) {
-        // Check if linearization state exists or need to be relinearized
-        if (not trkAtVtx.isLinearized || state.vtxInfoMap[vtx].relinearize) {
+        // Check if track is already linearized and whether we need to
+        // relinearize
+        if (not trkAtVtx.isLinearized or vtxInfo.relinearize) {
           auto result = linearizer.linearizeTrack(
-              m_extractParameters(*trk), state.vtxInfoMap[vtx].oldPosition[3],
+              m_extractParameters(*trk), vtxInfo.oldPosition[3],
               *vtxPerigeeSurface, vertexingOptions.geoContext,
               vertexingOptions.magFieldContext, state.linearizerState);
           if (!result.ok()) {
             return result.error();
           }
 
-          if (trkAtVtx.isLinearized) {
-            state.vtxInfoMap[vtx].linPoint = state.vtxInfoMap[vtx].oldPosition;
-          }
```

Comment:
Can you elaborate what this was doing before? It was updating the `linPoint` to what exactly? It's relinearizing here, if I read that right. So `linPoint` *is already* set to `oldPosition`? That's also what's passed to the linearization function right above.

Reply:
`oldPosition` is the vertex position from the last update of the fitter. I also found the naming a bit confusing!

In the refactor, I just pull this if-condition out of the loop, because it doesn't belong in there logically. The logic of the old code is:
If the track is not linearized yet or if we need to relinearize we perform linearization.
After the linearization, we check if the track was already linearized. If that is the case, that means we had to relinearize. Therefore, we need to update linPoint. 

I think that this is intricate (and also we set linPoint to the same value for each track). It is more logical to ask before the loop if the vertex requires relinearization, and, if it does, write:

![image](https://github.com/acts-project/acts/assets/72298366/92874167-f5e5-4b9d-b3e9-e9381015d9f5)

See a few lines above. Let me know if this is not clear or if I overlooked something!

Reply:
Edit: You are right that it does not make sense to use `oldPosition` in the linearization. I left it like this bc I was afraid it might change physmon and this was supposed to be a pure refactor!

I now use `linPoint`, let's see what happens :)